internal/tofu/eval_context_builtin.go

```diff
@@ -432,7 +433,25 @@ func (ctx *BuiltinEvalContext) EvaluationScope(self addrs.Referenceable, source
 	}
 
 	scope := ctx.Evaluator.Scope(data, self, source, func(pf addrs.ProviderFunction, rng tfdiags.SourceRange) (*function.Function, tfdiags.Diagnostics) {
-		return evalContextProviderFunction(ctx.Provider, mc, ctx.Evaluator.Operation, pf, rng)
+		absPc := ctx.ProviderFunctionTracker[ProviderFunctionReference{
```

Comment:
Nitpick: I'm not a super huge fan of looking up items in maps by using a large struct as a index as you are never sure how the equality is handled under the hood. I think I would prefer a simple `Lookup` or `Get` method on the ProviderFunctionTracker to handle something like this.

Reply:
[f39ad6e](https://github.com/opentofu/opentofu/pull/2082/commits/f39ad6e53fac0bd08461f3d903e146e9f8630f9d)